### ISSUE TITLE
convenience functions converting to/from xarray

### ISSUE THREAD

**Issue by aspitarl** [scipp/scipp#1849]
Hi, I'm was looking through the documentation and couldn't find this so I am just wondering: is there some sort of convenience method that can convert to and from xarray objects (with some amount of information loss possible). It seems like the data structures are similar enough that this should be somewhat straightforward like xarray does with the Dataset.from_dataframe method. I am able to accomplish this with the code below, but just wondering if I'm missing something or if this is in the radar of the package development. 

This code created a scipp DataArray from a xarray DataArray called `da_test` 

```
sc_coords = {coord : sc.Variable(dims=[coord], values=da_test.coords[coord].values) for coord in da_test.coords}

da_sc = sc.DataArray(
    data = sc.Variable(values= da_test.values, dims = da_test.dims),
    coords=sc_coords,
    attrs=da_test.attrs
    )
```

I'm just starting to play around with the scipp package, so sorry if I'm missing something basic.

**Comment by SimonHeybrock** [comment]
Hi Lee,
Have a look at `to_dict` and `from_dict` from https://scipp.github.io/python-reference/api.html#compatibility. I think at this point they may have certain limitations, but if you give it a try and tell us where it fails we can work on that.

I believe it is also not clear what should be done with units (maybe insert as attributes in xarray?) and variances.

**Comment by SimonHeybrock** [comment]
I gave it a try and run into trouble immediately. This does not work, because we use `values` where `xarray` expects `data`:

```python
import numpy as np
import scipp as sc
import xarray as xr
var = sc.array(dims=['x', 'y'], values=np.arange(6.0).reshape(3,2))
x = sc.array(dims=['x'], values=np.arange(3))
y = sc.array(dims=['y'], values=np.arange(2))
a = sc.DataArray(data=var, coords={'x':x, 'y':y})
xr.DataArray.from_dict(sc.to_dict(a))
```

Maybe we can consider changing `to_dict` to support "xarray style", e.g.,:
```python
sc.to_dict(a, style='xarray')
```
would
- drop (and warn) about variances
- drop (and warn) about masks
- change units to attrs
- drop scipp-style attrs (which are variables, in constrast to xarray, were attrs are just a dict of Python objects).

**Comment by aspitarl** [comment]
Yes, the functionality you describe in the list is what I was thinking of. Thanks for checking it out, I can just use the direct methods as I outlined for now. If I keep using the package this way perhaps I can try and integrate a function into to_dict as a possible contribution.
